packages/lib/src/components/ThreeDS2/components/Challenge/PrepareChallenge3DS2.tsx

```diff
@@ -60,19 +60,24 @@ class PrepareChallenge3DS2 extends Component<PrepareChallenge3DS2Props, PrepareC
     }
 
     public onActionHandled = (rtnObj: ActionHandledReturnObject) => {
-        this.props.onSubmitAnalytics({ type: THREEDS2_FULL, message: rtnObj.actionDescription });
+        this.props.onSubmitAnalytics({
+            type: THREEDS2_FULL,
+            message: rtnObj.actionDescription,
+            subtype: Analytics3DS2Events.CHALLENGE_IFRAME_LOADED
+        });
 
         this.props.onActionHandled(rtnObj);
     };
 
     public onFormSubmit = (msg: string) => {
         this.props.onSubmitAnalytics({
             type: THREEDS2_FULL,
-            message: msg
+            message: msg,
+            subtype: Analytics3DS2Events.CHALLENGE_DATA_SENT
         });
     };
 
-    setStatusComplete(resultObj) {
+    setStatusComplete(resultObj, isTimeout = false) {
```

Comment:
It feels like this function does too many things beside setting status to `complete`. 
It's not a blocker but we could put all the analytic logic inside the `componentDidUpdate` hook, and it would react to different state changes (potentially easy to extend the analytics logic in the future).